WordPress/AbstractFunctionRestrictionsSniff.php

```diff
@@ -224,10 +228,12 @@ public function is_targetted_token( $stackPtr ) {
 			}
 
 			if ( preg_match( $group['regex'], $token_content ) === 1 ) {
-				$this->process_matched_token( $stackPtr, $groupName, $token_content );
+				$skip_to = $this->process_matched_token( $stackPtr, $groupName, $token_content );
```

Comment:
> perhaps different sniffs would want different behaviors here.

Oh and I forgot to respond to that part. If we want to allow for that, I'd suggest splitting the `is_targetted_token()` method up into two, where the looping through the groups is moved to a separate method allowing child classes to overload just that particular part of the class.